app/views/devise/passwords/edit.html.erb

```diff
@@ -17,20 +17,20 @@ Unless required by applicable law or agreed to in writing, software distributed
   <h3 align='center'>Change your password</h3>
 
   <%= form_for(resource, as: resource_name, url: password_path(resource_name), html: { method: :put }) do |f| %>
```

Comment:
I wonder if changing this to `bootstrap_form_for` (which we use elsewhere in views) would automatically handle what you're doing with conditionally applying CSS classes.  See https://github.com/bootstrap-ruby/bootstrap_form#validation-and-errors for examples of configuring it wrt displaying errors.